server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -61,6 +61,10 @@ public String getName() {
     private final DiscoveryExtensionNode discoveryExtensionNode;
     private final TransportService transportService;
 
+    // Lists for filter headers
+    private List<String> allowList;
+    private List<String> denyList;
```

Comment:
This list of allowed/denied headers should be an Immutable collection and static - then the instance field can be removed.

Reply:
> This list of allowed/denied headers should be an Immutable collection and static - then the instance field can be removed.

Should I remove that two lines and define allowList, denyList in header filtering code or just add static and change type from List to Set:
`private static Set<String> allowList`
`private static Set<String> denyList`

Reply:
Just push a new code change, I'll resolve or provide more comments if need be.